coremltools/converters/mil/frontend/torch/torchir_passes.py

```diff
@@ -176,6 +166,13 @@ def _construct_nodes_to_fuse_inputs(nodes_to_fuse):
                 blocks=[],
             )
             graph.nodes[i] = tensor_assign_node
+        elif node.inputs:
+            node_input = node.inputs[0]
+            if node_input in tensor_to_node_sequence_mapping:
+                # update the mapping
+                node_output = node.outputs[0]
+                val = tensor_to_node_sequence_mapping[node_input]
+                tensor_to_node_sequence_mapping[node_output] = val
```

Comment:
Could you please share your reasoning on why this would be generally applicable for nodes in addition to `to`?

Also, what is the reason to delete line `del tensor_to_node_sequence_mapping[node_input]`? With that line, it makes `tensor_to_node_sequence_mapping[node_output]` to be ordered as latest

Reply:
> what is the reason to delete line del tensor_to_node_sequence_mapping[node_input]

with `del tensor_to_node_sequence_mapping[node_input]`, if cast source is referenced more than once, ValueError will be raised.

```
class Net(torch.nn.Module):
    def forward(self, x):
        y = torch.empty(x.shape)
        z = [y.to](http://y.to/)(torch.int32)
        w = [y.to](http://y.to/)(torch.int32)
        return z, w
```

stacktrace
```
Traceback (most recent call last):
  File "/Users/ryosukefukatani/work/coremltools/onth11.py", line 21, in <module>
    ct.TensorType("x", shape=(ct.RangeDim(), ct.RangeDim())),
  File "/Users/ryosukefukatani/work/coremltools/coremltools/converters/_converters_entry.py", line 542, in convert
    main_pipeline=pass_pipeline,
  File "/Users/ryosukefukatani/work/coremltools/coremltools/converters/mil/converter.py", line 188, in mil_convert
    return _mil_convert(model, convert_from, convert_to, ConverterRegistry, MLModel, compute_units, **kwargs)
  File "/Users/ryosukefukatani/work/coremltools/coremltools/converters/mil/converter.py", line 217, in _mil_convert
    **kwargs
  File "/Users/ryosukefukatani/work/coremltools/coremltools/converters/mil/converter.py", line 286, in mil_convert_to_proto
    prog = frontend_converter(model, **kwargs)
  File "/Users/ryosukefukatani/work/coremltools/coremltools/converters/mil/converter.py", line 108, in __call__
    return load(*args, **kwargs)
  File "/Users/ryosukefukatani/work/coremltools/coremltools/converters/mil/frontend/torch/load.py", line 61, in load
    specification_version,
  File "/Users/ryosukefukatani/work/coremltools/coremltools/converters/mil/frontend/torch/converter.py", line 335, in __init__
    p(self.graph)
  File "/Users/ryosukefukatani/work/coremltools/coremltools/converters/mil/frontend/torch/torchir_passes.py", line 141, in generate_tensor_assignment_ops
    raise ValueError("No matching select or slice.")
ValueError: No matching select or slice.
```

Reply:
> Could you please share your reasoning on why this would be generally applicable for nodes in addition to to?

We need general solution of #1917 

ex.
```
class Net(torch.nn.Module):
    def forward(self, x):
        y = torch.empty(x.shape) + 1
        y.fill_(0.0)
        return y
```

```
class Net(torch.nn.Module):
    def forward(self, x):
        y = torch.empty(x.shape) - 1
        y.fill_(0.0)
        return y
```


```
class Net(torch.nn.Module):
    def forward(self, x):
        y = torch.empty(x.shape) .flatten()
        y.fill_(0.0)
        return y
```

and all other operations.